### DISCUSSION TITLE
CommandLine.getCommands, positioning when onGetCommands is invoked.

### DISCUSSION THREAD

**Discussion by Jdance-Media** [SmartlyDressedGames/Unturned-3.x-Community#3948] [Ideas]
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/67397422/e6a9036c-b338-4781-aa4b-222a152112c6)
It would be extremely helpful if you could make it so onGetCommands is invoked after it gets all of the commands. I'd like to be able to rotate maps every server restart and this would make the process easier. My previous idea was to change Commands.dat when the server shuts down, but that seems pretty dumb.

__Why do you need access what commands were found?:__
Unturned server hosts such as LYHME always has the map command line added, leaving a blank value will result in it defaulting to PEI. So even if I added my own option using a module, LYHME's command will be executed afterwards, making my efforts useless.

**Comment by SDGNelson** [comment]
It's always interesting to see the decompiled code - in the source code there is no second list.

I'd like to rewrite the vanilla command stuff at some point.

In the meantime maybe Lyhme would be willing to remove that option when left empty?

**Reply by Jdance-Media** [reply]
Decompiler is weird like that sometimes. I'll check and see if LYHME would be willing to allow the field to be empty. 👍

**Comment by Jdance-Media** [comment]
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/67397422/3afe18f2-ebc3-42f1-858d-cb201dee43e4)
After doing some testing, I've realized that the ModuleHook would have to be started prior to Provider in order to modify the startup command lines. I'm not sure if OpenMod or RocketMod (LDM) relies on Provider awakening before they are initialized. This means that my idea was quite dumb.
